### DISCUSSION TITLE
How can I get the current element's ID in Kernel?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#22154] [Q&A General] [answered]
I found the method getElementIDIndex in [ElementIDInterface](https://mooseframework.inl.gov/source/interfaces/ElementIDInterface.html). But how to use it?

**Comment by hugary1995** [comment, marked as answer]
The doxygen documents are very useful tools to help you figure out this sort of question.

1. For almost everything related to element, you should consult the libmesh doxygen here: https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1Elem.html. The method you need is probably `Elem::id()`. There are other IDs of course.
2. For kernel objects, you should consult the MOOSE doxygen here e.g. https://mooseframework.inl.gov/docs/doxygen/moose/classKernel.html where you can find a member called `_current_elem` of type `const Elem *const &`.

Combining 1 and 2, you can use `_current_elem->id()` in any object that derives from `Kernel` to obtain the ID of the current element.

**Reply by Ethan-xj** [reply]
Thank you :D
